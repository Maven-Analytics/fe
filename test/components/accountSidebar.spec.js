import React from 'react';
import renderer from 'react-test-renderer';

import AccountSidebar from '../../src/components/accountSidebar';

describe('<AccountSidebar/>', () => {
  it('Should render without crashing', () => {
    const renderedComp = renderer.create(<AccountSidebar activeLink={0} />);

    let tree = renderedComp.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should log a warning to the console if activeLink prop is not passed in', () => {
    console.error = jest.fn();

    renderer.create(<AccountSidebar />);

    expect(console.error).toHaveBeenCalledTimes(1);
  });
});
