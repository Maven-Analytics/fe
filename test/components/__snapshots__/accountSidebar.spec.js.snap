// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountSidebar/> Should render without crashing 1`] = `
<div
  className="account-sidebar"
>
  <ul>
    <li>
      <a
        className="active"
        href="/dashboard/account"
        onClick={[Function]}
      >
        Profile
      </a>
    </li>
    <li>
      <a
        className=""
        href="/dashboard/account/password"
        onClick={[Function]}
      >
        Password
      </a>
    </li>
    <li>
      <a
        className=""
        href="//mavenanalytics.thinkific.com/account/billing"
        onClick={[Function]}
      >
        Billing
      </a>
    </li>
    <li>
      <a
        className=""
        href="//mavenanalytics.thinkific.com/account/orders"
        onClick={[Function]}
      >
        Order History
      </a>
    </li>
  </ul>
</div>
`;
